src/rust/src/encoder/common.rs

```diff
@@ -0,0 +1,545 @@
+const SIMPLE_XML_HEADER: &str = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<captions>\n";
+
+const WEBVTT_HEADER: &[&str] = &["WEBVTT\n", "\n", "\n"];
+
+const RCWT_HEADER: &[u8] = &[0xCC, 0xCC, 0xED, 0xCC, 0x00, 0x50, 0, 1, 0, 0, 0]; // "RCWT" + version
+
+pub fn encode_line(ctx: &mut encoder_ctx, buffer: &mut [c_uchar], text: &[u8]) -> c_uint {
+    if buffer.is_empty() {
+        return 0;
+    }
+
+    let mut bytes: c_uint = 0;
+    let mut buffer_pos = 0;
+    let mut text_pos = 0;
+    let text_len = text.iter().position(|&b| b == 0).unwrap_or(text.len());
+    while text_pos < text_len {
+        let current_byte = text[text_pos];
+
+        match ctx.encoding {
+            ccx_encoding_type_CCX_ENC_UTF_8 | ccx_encoding_type_CCX_ENC_LATIN_1 => {
+                if buffer_pos + 1 >= buffer.len() {
+                    break;
+                }
+
+                buffer[buffer_pos] = current_byte;
+                bytes += 1;
+                buffer_pos += 1;
+            }
+
+            ccx_encoding_type_CCX_ENC_UNICODE => {
+                if buffer_pos + 2 >= buffer.len() {
+                    break;
+                }
+
+                buffer[buffer_pos] = current_byte;
+                buffer[buffer_pos + 1] = 0;
+                bytes += 2;
+                buffer_pos += 2;
+            }
+            _ => {}
+        }
+        text_pos += 1;
+    }
+
+    // Add null terminator if there's space
+    if buffer_pos < buffer.len() {
+        buffer[buffer_pos] = 0;
+    }
+
+    bytes
+}
+pub fn write_subtitle_file_footer(ctx: &mut encoder_ctx, out: &mut ccx_s_write) -> c_int {
+    let mut ret: c_int = 0;
+    let mut str_buffer = [0u8; 1024];
+
+    match ctx.write_format {
+        ccx_output_format::CCX_OF_SAMI => {
+            let footer = b"</BODY></SAMI>\n\0";
```

Comment:
I found out that for CCX_OF_SAMI, CCX_OF_SMPTETT, CCX_OF_SIMPLE_XML further logic is same only footer is different. So could you combine these cases into one and only change the value of footer based on which one we are in?